### ISSUE TITLE
Keras model setup (miniconda) in RStudio

### ISSUE THREAD

**Issue by Little-Miki** [rstudio/keras#1284]
I am attempting to create a simple keras model in R Studio, using ```model <- keras_model_sequential```

I have the following libraries install and present: ```keras``` ```dplyr``` ```reticulate```

However, when I run the initial ```model``` code, I receive the following error:

> Error in conda_python(envpath, conda = miniconda) : no conda environment exists at path 'C:/Users/.../AppData/Local/r-miniconda/envs/r-reticulate'

Or

> Error: Error 1 occurred creating conda environment C:/Users/.../AppData/Local/r-miniconda/envs/r-reticulate

I have tried uninstalling miniconda and reinstalling an older version (4.5.11) as I read that that may help.

I have also unchecked the "Use secure method for HTTP" and "Use Internet Explorer library/proxy for HTTP" as I also read that that may help.

I have even tried doing this on a different machine and get the same error.

I'm unsure as to how to fix this error and I'm lost now!

Any help will be appreciated.

**Comment by t-kalinowski** [comment]
Can you please post the full output from running:
```
install.github("remotes")
# upgrade to the current development version to make sure the issue is still present
remotes::install_github(c("rstudio/reticulate", "rstudio/tensorflow", "rstudio/keras")

# delete previous miniconda installation so we start from a fresh state
unlink(reticulate::miniconda_path(), recursive = TRUE)

# install python, then tensorflow+keras
reticulate::install_miniconda()
keras::install_keras()

# test if Tensorflow installed successfully
tensorflow::tf$abs(1)

# if the above failed, inspect the actual import error:
reticulate::import("tensorflow")
for(msg in reticulate::py_last_error()) cat(msg, "\n")
```

**Comment by Little-Miki** [comment]
install.github("remotes")
Error in install.github("remotes") : 
  could not find function "install.github"
> remotes::install_github(c("rstudio/reticulate", "rstudio/tensorflow", "rstudio/keras"))
Downloading GitHub repo rstudio/reticulate@HEAD
√  checking for file 'C:\Users\C4094646\AppData\Local\Temp\RtmpQba2Zs\remotes22a81edb47e5\rstudio-reticulate-3eda565/DESCRIPTION' (438ms)
-  preparing 'reticulate': (2.3s)
√  checking DESCRIPTION meta-information ... 
-  cleaning src
-  checking for LF line-endings in source and make files and shell scripts (491ms)
-  checking for empty or unneeded directories
-  building 'reticulate_1.22-9000.tar.gz'
   
Installing package into ‘C:/Users/C4094646/Documents/R/win-library/4.0’
(as ‘lib’ is unspecified)
* installing *source* package 'reticulate' ...
ERROR: cannot remove earlier installation, is it in use?
* removing 'C:/Users/C4094646/Documents/R/win-library/4.0/reticulate'
* restoring previous 'C:/Users/C4094646/Documents/R/win-library/4.0/reticulate'
Warning in file.copy(lp, dirname(pkgdir), recursive = TRUE, copy.date = TRUE) :
  problem copying C:\Users\C4094646\Documents\R\win-library\4.0\00LOCK-reticulate\reticulate\libs\x64\reticulate.dll to C:\Users\C4094646\Documents\R\win-library\4.0\reticulate\libs\x64\reticulate.dll: Permission denied
Downloading GitHub repo rstudio/tensorflow@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                            
2: CRAN packages only             
3: None                           
4: rlang (0.4.11 -> 0.4.12) [CRAN]

Enter one or more numbers, or an empty line to skip updates: 1
rlang (0.4.11 -> 0.4.12) [CRAN]
Installing 1 packages: rlang
Installing package into ‘C:/Users/C4094646/Documents/R/win-library/4.0’
(as ‘lib’ is unspecified)
trying URL 'http://cran.rstudio.com/bin/windows/contrib/4.0/rlang_0.4.12.zip'
Content type 'application/zip' length 1200906 bytes (1.1 MB)
downloaded 1.1 MB

package ‘rlang’ successfully unpacked and MD5 sums checked
Warning: cannot remove prior installation of package ‘rlang’
Warning: restored ‘rlang’

The downloaded binary packages are in
	C:\Users\C4094646\AppData\Local\Temp\RtmpQba2Zs\downloaded_packages
√  checking for file 'C:\Users\C4094646\AppData\Local\Temp\RtmpQba2Zs\remotes22a8732f3917\rstudio-tensorflow-dbe0403/DESCRIPTION' (428ms)
-  preparing 'tensorflow': (635ms)
√  checking DESCRIPTION meta-information ... 
-  checking for LF line-endings in source and make files and shell scripts
-  checking for empty or unneeded directories
-  building 'tensorflow_2.6.0.9000.tar.gz'
   
Installing package into ‘C:/Users/C4094646/Documents/R/win-library/4.0’
(as ‘lib’ is unspecified)
* installing *source* package 'tensorflow' ...
** using staged installation
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
  converting help for package 'tensorflow'
    finding HTML links ... done
    all_dims                                html  
    as_tensor                               html  
    evaluate                                html  
    export_savedmodel                       html  
    install_tensorflow                      html  
    install_tensorflow_extras               html  
    parse_arguments                         html  
    finding level-2 HTML links ... done

    parse_flags                             html  
    reexports                               html  
    set_random_seed                         html  
    shape                                   html  
    sub-.tensorflow.tensor                  html  
    tensorboard                             html  
    tensorflow                              html  
    tf                                      html  
    tf_config                               html  
    tf_extract_opts                         html  
    tf_function                             html  
    tf_gpu_configured                       html  
    tf_probability                          html  
    tfe_enable_eager_execution              html  
    train                                   html  
    train_and_evaluate                      html  
    use_compat                              html  
    use_session_with_seed                   html  
    view_savedmodel                         html  
** building package indices
** testing if installed package can be loaded from temporary location
*** arch - i386
*** arch - x64
** testing if installed package can be loaded from final location
*** arch - i386
*** arch - x64
** testing if installed package keeps a record of temporary installation path
* DONE (tensorflow)
Downloading GitHub repo rstudio/keras@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                            
2: CRAN packages only             
3: None                           
4: rlang (0.4.11 -> 0.4.12) [CRAN]

Enter one or more numbers, or an empty line to skip updates: 1
rlang (0.4.11 -> 0.4.12) [CRAN]
Installing 1 packages: rlang
Installing package into ‘C:/Users/C4094646/Documents/R/win-library/4.0’
(as ‘lib’ is unspecified)
trying URL 'http://cran.rstudio.com/bin/windows/contrib/4.0/rlang_0.4.12.zip'
Content type 'application/zip' length 1200906 bytes (1.1 MB)
downloaded 1.1 MB

package ‘rlang’ successfully unpacked and MD5 sums checked
Warning: cannot remove prior installation of package ‘rlang’
Warning: restored ‘rlang’

The downloaded binary packages are in
	C:\Users\C4094646\AppData\Local\Temp\RtmpQba2Zs\downloaded_packages
√  checking for file 'C:\Users\C4094646\AppData\Local\Temp\RtmpQba2Zs\remotes22a82db95d6d\rstudio-keras-4be23b2/DESCRIPTION' (480ms)
-  preparing 'keras': (15.6s)
√  checking DESCRIPTION meta-information ... 
-  checking for LF line-endings in source and make files and shell scripts (4.1s)
-  checking for empty or unneeded directories
   Removed empty directory 'keras/man-roxygen'
-  building 'keras_2.6.1.9000.tar.gz'
   
Installing package into ‘C:/Users/C4094646/Documents/R/win-library/4.0’
(as ‘lib’ is unspecified)
* installing *source* package 'keras' ...
** using staged installation
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
  converting help for package 'keras'
    finding HTML links ... done
    KerasCallback                           html  
    finding level-2 HTML links ... done

    KerasConstraint                         html  
    KerasLayer                              html  
    KerasWrapper                            html  
    Layer                                   html  
    Metric                                  html  
    activation_relu                         html  
    adapt                                   html  
    application_densenet                    html  
    application_efficientnet                html  
    application_inception_resnet_v2         html  
    application_inception_v3                html  
    application_mobilenet                   html  
    application_mobilenet_v2                html  
    application_mobilenet_v3                html  
    application_nasnet                      html  
    application_resnet                      html  
    application_vgg                         html  
    application_xception                    html  
    backend                                 html  
    bidirectional                           html  
    callback_csv_logger                     html  
    callback_early_stopping                 html  
    callback_lambda                         html  
    callback_learning_rate_scheduler        html  
    callback_model_checkpoint               html  
    callback_progbar_logger                 html  
    callback_reduce_lr_on_plateau           html  
    callback_remote_monitor                 html  
    callback_tensorboard                    html  
    callback_terminate_on_naan              html  
    clone_model                             html  
    compile.keras.engine.training.Model     html  
    constraints                             html  
    count_params                            html  
    create_layer                            html  
    create_layer_wrapper                    html  
    create_wrapper                          html  
    custom_metric                           html  
    dataset_boston_housing                  html  
    dataset_cifar10                         html  
    dataset_cifar100                        html  
    dataset_fashion_mnist                   html  
    dataset_imdb                            html  
    dataset_mnist                           html  
    dataset_reuters                         html  
    evaluate.keras.engine.training.Model    html  
    evaluate_generator                      html  
    export_savedmodel.keras.engine.training.Model
                                            html  
    fit.keras.engine.training.Model         html  
    fit_generator                           html  
    fit_image_data_generator                html  
    fit_text_tokenizer                      html  
    flow_images_from_data                   html  
    flow_images_from_dataframe              html  
    flow_images_from_directory              html  
    freeze_weights                          html  
    generator_next                          html  
    get_config                              html  
    get_file                                html  
    get_input_at                            html  
    get_layer                               html  
    get_weights                             html  
    grapes-py_class-grapes                  html  
    grapes-set-active-grapes                html  
    hdf5_matrix                             html  
    image_data_generator                    html  
    image_dataset_from_directory            html  
    image_load                              html  
    image_to_array                          html  
    imagenet_decode_predictions             html  
    imagenet_preprocess_input               html  
    implementation                          html  
    initializer_constant                    html  
    initializer_glorot_normal               html  
    initializer_glorot_uniform              html  
    initializer_he_normal                   html  
    initializer_he_uniform                  html  
    initializer_identity                    html  
    initializer_lecun_normal                html  
    initializer_lecun_uniform               html  
    initializer_ones                        html  
    initializer_orthogonal                  html  
    initializer_random_normal               html  
    initializer_random_uniform              html  
    initializer_truncated_normal            html  
    initializer_variance_scaling            html  
    initializer_zeros                       html  
    install_keras                           html  
    is_keras_available                      html  
    k_abs                                   html  
    k_all                                   html  
    k_any                                   html  
    k_arange                                html  
    k_argmax                                html  
    k_argmin                                html  
    k_backend                               html  
    k_batch_dot                             html  
    k_batch_flatten                         html  
    k_batch_get_value                       html  
    k_batch_normalization                   html  
    k_batch_set_value                       html  
    k_bias_add                              html  
    k_binary_crossentropy                   html  
    k_cast                                  html  
    k_cast_to_floatx                        html  
    k_categorical_crossentropy              html  
    k_clear_session                         html  
    k_clip                                  html  
    k_concatenate                           html  
    k_constant                              html  
    k_conv1d                                html  
    k_conv2d                                html  
    k_conv2d_transpose                      html  
    k_conv3d                                html  
    k_conv3d_transpose                      html  
    k_cos                                   html  
    k_count_params                          html  
    k_ctc_batch_cost                        html  
    k_ctc_decode                            html  
    k_ctc_label_dense_to_sparse             html  
    k_cumprod                               html  
    k_cumsum                                html  
    k_depthwise_conv2d                      html  
    k_dot                                   html  
    k_dropout                               html  
    k_dtype                                 html  
    k_elu                                   html  
    k_epsilon                               html  
    k_equal                                 html  
    k_eval                                  html  
    k_exp                                   html  
    k_expand_dims                           html  
    k_eye                                   html  
    k_flatten                               html  
    k_floatx                                html  
    k_foldl                                 html  
    k_foldr                                 html  
    k_function                              html  
    k_gather                                html  
    k_get_session                           html  
    k_get_uid                               html  
    k_get_value                             html  
    k_get_variable_shape                    html  
    k_gradients                             html  
    k_greater                               html  
    k_greater_equal                         html  
    k_hard_sigmoid                          html  
    k_identity                              html  
    k_image_data_format                     html  
    k_in_test_phase                         html  
    k_in_top_k                              html  
    k_in_train_phase                        html  
    k_int_shape                             html  
    k_is_keras_tensor                       html  
    k_is_placeholder                        html  
    k_is_sparse                             html  
    k_is_tensor                             html  
    k_l2_normalize                          html  
    k_learning_phase                        html  
    k_less                                  html  
    k_less_equal                            html  
    k_local_conv1d                          html  
    k_local_conv2d                          html  
    k_log                                   html  
    k_logsumexp                             html  
    k_manual_variable_initialization        html  
    k_map_fn                                html  
    k_max                                   html  
    k_maximum                               html  
    k_mean                                  html  
    k_min                                   html  
    k_minimum                               html  
    k_moving_average_update                 html  
    k_ndim                                  html  
    k_normalize_batch_in_training           html  
    k_not_equal                             html  
    k_one_hot                               html  
    k_ones                                  html  
    k_ones_like                             html  
    k_permute_dimensions                    html  
    k_placeholder                           html  
    k_pool2d                                html  
    k_pool3d                                html  
    k_pow                                   html  
    k_print_tensor                          html  
    k_prod                                  html  
    k_random_binomial                       html  
    k_random_normal                         html  
    k_random_normal_variable                html  
    k_random_uniform                        html  
    k_random_uniform_variable               html  
    k_relu                                  html  
    k_repeat                                html  
    k_repeat_elements                       html  
    k_reset_uids                            html  
    k_reshape                               html  
    k_resize_images                         html  
    k_resize_volumes                        html  
    k_reverse                               html  
    k_rnn                                   html  
    k_round                                 html  
    k_separable_conv2d                      html  
    k_set_learning_phase                    html  
    k_set_value                             html  
    k_shape                                 html  
    k_sigmoid                               html  
    k_sign                                  html  
    k_sin                                   html  
    k_softmax                               html  
    k_softplus                              html  
    k_softsign                              html  
    k_sparse_categorical_crossentropy       html  
    k_spatial_2d_padding                    html  
    k_spatial_3d_padding                    html  
    k_sqrt                                  html  
    k_square                                html  
    k_squeeze                               html  
    k_stack                                 html  
    k_std                                   html  
    k_stop_gradient                         html  
    k_sum                                   html  
    k_switch                                html  
    k_tanh                                  html  
    k_temporal_padding                      html  
    k_tile                                  html  
    k_to_dense                              html  
    k_transpose                             html  
    k_truncated_normal                      html  
    k_update                                html  
    k_update_add                            html  
    k_update_sub                            html  
    k_var                                   html  
    k_variable                              html  
    k_zeros                                 html  
    k_zeros_like                            html  
    keras-package                           html  
    keras                                   html  
    keras_array                             html  
    keras_model                             html  
    keras_model_custom                      html  
    keras_model_sequential                  html  
    layer_activation                        html  
    layer_activation_elu                    html  
    layer_activation_leaky_relu             html  
    layer_activation_parametric_relu        html  
    layer_activation_relu                   html  
    layer_activation_selu                   html  
    layer_activation_softmax                html  
    layer_activation_thresholded_relu       html  
    layer_activity_regularization           html  
    layer_add                               html  
    layer_additive_attention                html  
    layer_alpha_dropout                     html  
    layer_attention                         html  
    layer_average                           html  
    layer_average_pooling_1d                html  
    layer_average_pooling_2d                html  
    layer_average_pooling_3d                html  
    layer_batch_normalization               html  
    layer_category_encoding                 html  
    layer_center_crop                       html  
    layer_concatenate                       html  
    layer_conv_1d                           html  
    layer_conv_1d_transpose                 html  
    layer_conv_2d                           html  
    layer_conv_2d_transpose                 html  
    layer_conv_3d                           html  
    layer_conv_3d_transpose                 html  
    layer_conv_lstm_1d                      html  
    layer_conv_lstm_2d                      html  
    layer_conv_lstm_3d                      html  
    layer_cropping_1d                       html  
    layer_cropping_2d                       html  
    layer_cropping_3d                       html  
    layer_cudnn_gru                         html  
    layer_cudnn_lstm                        html  
    layer_dense                             html  
    layer_dense_features                    html  
    layer_depthwise_conv_2d                 html  
    layer_discretization                    html  
    layer_dot                               html  
    layer_dropout                           html  
    layer_embedding                         html  
    layer_flatten                           html  
    layer_gaussian_dropout                  html  
    layer_gaussian_noise                    html  
    layer_global_average_pooling_1d         html  
    layer_global_average_pooling_2d         html  
    layer_global_average_pooling_3d         html  
    layer_global_max_pooling_1d             html  
    layer_global_max_pooling_2d             html  
    layer_global_max_pooling_3d             html  
    layer_gru                               html  
    layer_gru_cell                          html  
    layer_hashing                           html  
    layer_input                             html  
    layer_integer_lookup                    html  
    layer_lambda                            html  
    layer_layer_normalization               html  
    layer_locally_connected_1d              html  
    layer_locally_connected_2d              html  
    layer_lstm                              html  
    layer_lstm_cell                         html  
    layer_masking                           html  
    layer_max_pooling_1d                    html  
    layer_max_pooling_2d                    html  
    layer_max_pooling_3d                    html  
    layer_maximum                           html  
    layer_minimum                           html  
    layer_multi_head_attention              html  
    layer_multiply                          html  
    layer_normalization                     html  
    layer_permute                           html  
    layer_random_contrast                   html  
    layer_random_crop                       html  
    layer_random_flip                       html  
    layer_random_height                     html  
    layer_random_rotation                   html  
    layer_random_translation                html  
    layer_random_width                      html  
    layer_random_zoom                       html  
    layer_repeat_vector                     html  
    layer_rescaling                         html  
    layer_reshape                           html  
    layer_resizing                          html  
    layer_rnn                               html  
    layer_separable_conv_1d                 html  
    layer_separable_conv_2d                 html  
    layer_simple_rnn                        html  
    layer_simple_rnn_cell                   html  
    layer_spatial_dropout_1d                html  
    layer_spatial_dropout_2d                html  
    layer_spatial_dropout_3d                html  
    layer_stacked_rnn_cells                 html  
    layer_string_lookup                     html  
    layer_subtract                          html  
    layer_text_vectorization                html  
    layer_upsampling_1d                     html  
    layer_upsampling_2d                     html  
    layer_upsampling_3d                     html  
    layer_zero_padding_1d                   html  
    layer_zero_padding_2d                   html  
    layer_zero_padding_3d                   html  
    loss-functions                          html  
    loss_cosine_proximity                   html  
    make_sampling_table                     html  
    metric-or-Metric                        html  
    metric_accuracy                         html  
    metric_auc                              html  
    metric_binary_accuracy                  html  
    metric_binary_crossentropy              html  
    metric_categorical_accuracy             html  
    metric_categorical_crossentropy         html  
    metric_categorical_hinge                html  
    metric_cosine_proximity                 html  
    metric_cosine_similarity                html  
    metric_false_negatives                  html  
    metric_false_positives                  html  
    metric_hinge                            html  
    metric_kullback_leibler_divergence      html  
    metric_logcosh_error                    html  
    metric_mean                             html  
    metric_mean_absolute_error              html  
    metric_mean_absolute_percentage_error   html  
    metric_mean_iou                         html  
    metric_mean_relative_error              html  
    metric_mean_squared_error               html  
    metric_mean_squared_logarithmic_error   html  
    metric_mean_tensor                      html  
    metric_mean_wrapper                     html  
    metric_poisson                          html  
    metric_precision                        html  
    metric_precision_at_recall              html  
    metric_recall                           html  
    metric_recall_at_precision              html  
    metric_root_mean_squared_error          html  
    metric_sensitivity_at_specificity       html  
    metric_sparse_categorical_accuracy      html  
    metric_sparse_categorical_crossentropy
                                            html  
    metric_sparse_top_k_categorical_accuracy
                                            html  
    metric_specificity_at_sensitivity       html  
    metric_squared_hinge                    html  
    metric_sum                              html  
    metric_top_k_categorical_accuracy       html  
    metric_true_negatives                   html  
    metric_true_positives                   html  
    model_from_saved_model                  html  
    model_to_json                           html  
    model_to_saved_model                    html  
    model_to_yaml                           html  
    multi-assign                            html  
Rd warning: C:/Users/C4094646/AppData/Local/Temp/RtmpAJnwxI/R.INSTALL2bc8249767b5/keras/man/multi-assign.Rd:10: file link '%<-%' in package 'zeallot' does not exist and so has been treated as a topic
    multi_gpu_model                         html  
    normalize                               html  
    optimizer_adadelta                      html  
    optimizer_adagrad                       html  
    optimizer_adam                          html  
    optimizer_adamax                        html  
    optimizer_nadam                         html  
    optimizer_rmsprop                       html  
    optimizer_sgd                           html  
    pad_sequences                           html  
    pipe                                    html  
Rd warning: C:/Users/C4094646/AppData/Local/Temp/RtmpAJnwxI/R.INSTALL2bc8249767b5/keras/man/pipe.Rd:10: file link '%>%' in package 'magrittr' does not exist and so has been treated as a topic
    plot.keras_training_history             html  
    pop_layer                               html  
    predict.keras.engine.training.Model     html  
    predict_generator                       html  
    predict_on_batch                        html  
    predict_proba                           html  
    reexports                               html  
    regularizer_l1                          html  
    reset_states                            html  
    save_model_hdf5                         html  
    save_model_tf                           html  
    save_model_weights_hdf5                 html  
    save_model_weights_tf                   html  
    save_text_tokenizer                     html  
    sequences_to_matrix                     html  
    sequential_model_input_layer            html  
    serialize_model                         html  
    skipgrams                               html  
    summary.keras.engine.training.Model     html  
    text_dataset_from_directory             html  
    text_hashing_trick                      html  
    text_one_hot                            html  
    text_to_word_sequence                   html  
    text_tokenizer                          html  
    texts_to_matrix                         html  
    texts_to_sequences                      html  
    texts_to_sequences_generator            html  
    time_distributed                        html  
    timeseries_generator                    html  
    to_categorical                          html  
    train_on_batch                          html  
    use_implementation                      html  
    with_custom_object_scope                html  
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
*** arch - i386
*** arch - x64
** testing if installed package can be loaded from final location
*** arch - i386
*** arch - x64
** testing if installed package keeps a record of temporary installation path
* DONE (keras)
Warning messages:
1: In i.p(...) :
  installation of package ‘C:/Users/C4094646/AppData/Local/Temp/RtmpQba2Zs/file22a872c6630d/reticulate_1.22-9000.tar.gz’ had non-zero exit status
2: In file.copy(savedcopy, lib, recursive = TRUE) :
  problem copying C:\Users\C4094646\Documents\R\win-library\4.0\00LOCK\rlang\libs\x64\rlang.dll to C:\Users\C4094646\Documents\R\win-library\4.0\rlang\libs\x64\rlang.dll: Permission denied
3: In file.copy(savedcopy, lib, recursive = TRUE) :
  problem copying C:\Users\C4094646\Documents\R\win-library\4.0\00LOCK\rlang\libs\x64\rlang.dll to C:\Users\C4094646\Documents\R\win-library\4.0\rlang\libs\x64\rlang.dll: Permission denied
> unlink(reticulate::miniconda_path(), recursive = TRUE)
> reticulate::install_miniconda()
* Downloading "https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe" ...
trying URL 'https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe'
Content type 'application/octet-stream' length 60924672 bytes (58.1 MB)
downloaded 58.1 MB

* Installing Miniconda -- please wait a moment ...
This program is blocked by group policy. For more information, contact your system administrator.
This program is blocked by group policy. For more information, contact your system administrator.
Error: Error 1 occurred creating conda environment r-reticulate
> tensorflow::tf$abs(1)
This program is blocked by group policy. For more information, contact your system administrator.
This program is blocked by group policy. For more information, contact your system administrator.
Error: Error 1 occurred creating conda environment C:/Users/C4094646/AppData/Local/r-miniconda/envs/r-reticulate
> reticulate::import("tensorflow")
This program is blocked by group policy. For more information, contact your system administrator.
Error: Error 1 occurred creating conda environment C:/Users/C4094646/AppData/Local/r-miniconda/envs/r-reticulate
> for(msg in reticulate::py_last_error()) cat(msg, "\n")

**Comment by Little-Miki** [comment]
I will also run this on another machine later on to see if I get anything different, as permissions maybe an issue on the machine I am currently using. However, if you have any idea what's going on, that would be great.

**Comment by t-kalinowski** [comment]
Hi, it looks you're unable to download and install miniconda and/or create conda environments on your machine due to the configured permissions. You'll have to get in touch with your system administrator to allow installation. (Your experience might be different if you run R from WSL.)

**Comment by Little-Miki** [comment]
I am using a machine from my workplace. I have run this on my own personal machine and get the same initial error. I will run your code later on my own personal machine were there are no permissions issues.

**Comment by Little-Miki** [comment]
`install.packages("remotes")`
>WARNING: Rtools is required to build R packages but is not currently installed. Please download and install the appropriate version of Rtools before proceeding:
>
>https://cran.rstudio.com/bin/windows/Rtools/
Installing package into ‘C:/Users/Mikip/OneDrive/Documents/R/win-library/4.1’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.1/remotes_2.4.1.zip'
Content type 'application/zip' length 399339 bytes (389 KB)
downloaded 389 KB
>
>package ‘remotes’ successfully unpacked and MD5 sums checked
>
>The downloaded binary packages are in
	C:\Users\Mikip\AppData\Local\Temp\RtmpklyGXr\downloaded_packages

`# upgrade to the current development version to make sure the issue is still present`
`remotes::install_github(c("rstudio/reticulate", "rstudio/tensorflow", "rstudio/keras"))`
>Downloading GitHub repo rstudio/reticulate@HEAD
Running `R CMD build`...
>* checking for file 'C:\Users\Mikip\AppData\Local\Temp\RtmpklyGXr\remotes23445a805389\rstudio-reticulate-3eda565/DESCRIPTION' ... OK
>* preparing 'reticulate':
>* checking DESCRIPTION meta-information ... OK
>* cleaning src
>* checking for LF line-endings in source and make files and shell scripts
>* checking for empty or unneeded directories
>* building 'reticulate_1.22-9000.tar.gz'
>Installing package into ‘C:/Users/Mikip/OneDrive/Documents/R/win-library/4.1’
(as ‘lib’ is unspecified)
>* installing *source* package 'reticulate' ...
>** using staged installation
>** libs
>Warning in system(cmd) : 'make' not found
>
>ERROR: compilation failed for package 'reticulate'
>* removing 'C:/Users/Mikip/OneDrive/Documents/R/win-library/4.1/reticulate'
>* restoring previous 'C:/Users/Mikip/OneDrive/Documents/R/win-library/4.1/reticulate'
>Skipping install of 'tensorflow' from a github remote, the SHA1 (dbe0403b) has not changed since last install.
>  Use `force = TRUE` to force installation
>Skipping install of 'keras' from a github remote, the SHA1 (4be23b21) has not changed since last install.
>  Use `force = TRUE` to force installation
>
>Warning messages:
>1: In missing_devel_warning(pkgdir) :
>  Package reticulate has compiled code, but no suitable compiler(s) were found. Installation will likely fail.
>  Install Rtools (https://cran.r-project.org/bin/windows/Rtools/).Then use the pkgbuild package, or make sure that Rtools in the PATH.
>2: In i.p(...) :
>  installation of package ‘C:/Users/Mikip/AppData/Local/Temp/RtmpklyGXr/file234446fd687f/reticulate_1.22-9000.tar.gz’ had non-zero exit status
 
`# delete previous miniconda installation so we start from a fresh state`
`unlink(reticulate::miniconda_path(), recursive = TRUE)`

`# install python, then tensorflow+keras`
`reticulate::install_miniconda()`
>Downloading "https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe" ...
trying URL 'https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe'
Content type 'application/octet-stream' length 60924672 bytes (58.1 MB)
downloaded 58.1 MB
>
>* Installing Miniconda -- please wait a moment ...
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... done
>
>## Package Plan ##
>
>  environment location: C:\Users\Mikip\AppData\Local\R-MINI~1
>
>  added / updated specs:
 >   - conda
>
>The following packages will be downloaded:
>
>    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2021.10.26 |       haa95532_2         115 KB
    certifi-2021.10.8          |   py39haa95532_0         152 KB
    charset-normalizer-2.0.4   |     pyhd3eb1b0_0          35 KB
    cryptography-35.0.0        |   py39h71e12ea_0         991 KB
    idna-3.2                   |     pyhd3eb1b0_0          48 KB
    openssl-1.1.1l             |       h2bbff1b_0         4.8 MB
    pip-21.2.4                 |   py39haa95532_0         1.8 MB
    pyopenssl-21.0.0           |     pyhd3eb1b0_1          49 KB
    pywin32-228                |   py39hbaba5e8_1         5.6 MB
    requests-2.26.0            |     pyhd3eb1b0_0          59 KB
    setuptools-58.0.4          |   py39haa95532_0         778 KB
    tqdm-4.62.3                |     pyhd3eb1b0_1          83 KB
    tzdata-2021e               |       hda174b7_0         112 KB
    urllib3-1.26.7             |     pyhd3eb1b0_0         111 KB
    wheel-0.37.0               |     pyhd3eb1b0_1          33 KB
    wincertstore-0.2           |   py39haa95532_2          15 KB
    ------------------------------------------------------------
                                           Total:        14.8 MB
>
>The following NEW packages will be INSTALLED:
>
>  charset-normalizer pkgs/main/noarch::charset-normalizer-2.0.4-pyhd3eb1b0_0
>
>The following packages will be REMOVED:
>
>  chardet-4.0.0-py39haa95532_1003
>
>The following packages will be UPDATED:
>
>  ca-certificates                       2021.7.5-haa95532_1 --> 2021.10.26-haa95532_2
>  certifi                          2021.5.30-py39haa95532_0 --> 2021.10.8-py39haa95532_0
>  cryptography                         3.4.7-py39h71e12ea_0 --> 35.0.0-py39h71e12ea_0
>  idna                                    2.10-pyhd3eb1b0_0 --> 3.2-pyhd3eb1b0_0
>  openssl                                 1.1.1k-h2bbff1b_0 --> 1.1.1l-h2bbff1b_0
>  pip                                 21.1.3-py39haa95532_0 --> 21.2.4-py39haa95532_0
>  pyopenssl                             20.0.1-pyhd3eb1b0_1 --> 21.0.0-pyhd3eb1b0_1
>  pywin32                                228-py39he774522_0 --> 228-py39hbaba5e8_1
>  requests                              2.25.1-pyhd3eb1b0_0 --> 2.26.0-pyhd3eb1b0_0
>  setuptools                          52.0.0-py39haa95532_0 --> 58.0.4-py39haa95532_0
>  tqdm                                  4.61.2-pyhd3eb1b0_1 --> 4.62.3-pyhd3eb1b0_1
>  tzdata                                   2021a-h52ac0ba_0 --> 2021e-hda174b7_0
>  urllib3                               1.26.6-pyhd3eb1b0_1 --> 1.26.7-pyhd3eb1b0_0
>  wheel                                 0.36.2-pyhd3eb1b0_0 --> 0.37.0-pyhd3eb1b0_1
>  wincertstore                           0.2-py39h2bbff1b_0 --> 0.2-py39haa95532_2
>
>
>
>Downloading and Extracting Packages
>tzdata-2021e         | 112 KB    | ########## | 100% 
>certifi-2021.10.8    | 152 KB    | ########## | 100% 
>cryptography-35.0.0  | 991 KB    | ########## | 100% 
>setuptools-58.0.4    | 778 KB    | ########## | 100% 
>pyopenssl-21.0.0     | 49 KB     | ########## | 100% 
>openssl-1.1.1l       | 4.8 MB    | ########## | 100% 
>ca-certificates-2021 | 115 KB    | ########## | 100% 
>requests-2.26.0      | 59 KB     | ########## | 100% 
>wincertstore-0.2     | 15 KB     | ########## | 100% 
>wheel-0.37.0         | 33 KB     | ########## | 100% 
>charset-normalizer-2 | 35 KB     | ########## | 100% 
>tqdm-4.62.3          | 83 KB     | ########## | 100% 
>urllib3-1.26.7       | 111 KB    | ########## | 100% 
>idna-3.2             | 48 KB     | ########## | 100% 
>pip-21.2.4           | 1.8 MB    | ########## | 100% 
>pywin32-228          | 5.6 MB    | ########## | 100% 
>Preparing transaction: ...working... done
>Verifying transaction: ...working... done
>Executing transaction: ...working... done
>Collecting package metadata (current_repodata.json): ...working... done
>Solving environment: ...working... done
>
>## Package Plan ##
>
>  environment location: C:\Users\Mikip\AppData\Local\R-MINI~1\envs\r-reticulate
>
>  added / updated specs:
>    - numpy
>    - python=3.6
>
>
>The following packages will be downloaded:
>
>    package                    |            build
    ---------------------------|-----------------
    certifi-2020.12.5          |   py36haa95532_0         140 KB
    intel-openmp-2021.4.0      |    h57928b3_3556         3.2 MB  conda-forge
    libblas-3.9.0              |     12_win64_mkl         4.5 MB  conda-forge
    libcblas-3.9.0             |     12_win64_mkl         4.5 MB  conda-forge
    liblapack-3.9.0            |     12_win64_mkl         4.5 MB  conda-forge
    mkl-2021.4.0               |     h0e2418a_729       181.7 MB  conda-forge
    numpy-1.19.5               |   py36h4b40d73_2         4.9 MB  conda-forge
    pip-21.3.1                 |     pyhd8ed1ab_0         1.2 MB  conda-forge
    python-3.6.13              |h39d44d4_2_cpython        18.9 MB  conda-forge
    python_abi-3.6             |          2_cp36m           4 KB  conda-forge
    setuptools-49.6.0          |   py36ha15d459_3         921 KB  conda-forge
    tbb-2021.4.0               |       h2d74725_1         149 KB  conda-forge
    ucrt-10.0.20348.0          |       h57928b3_0         1.2 MB  conda-forge
    vc-14.2                    |       hb210afc_5          13 KB  conda-forge
    vs2015_runtime-14.29.30037 |       h902a5da_5         1.3 MB  conda-forge
    wheel-0.37.0               |     pyhd8ed1ab_1          31 KB  conda-forge
    wincertstore-0.2           |py36ha15d459_1006          15 KB  conda-forge
    ------------------------------------------------------------
                                           Total:       227.0 MB
>
>The following NEW packages will be INSTALLED:
>
>  certifi            pkgs/main/win-64::certifi-2020.12.5-py36haa95532_0
>  intel-openmp       conda-forge/win-64::intel-openmp-2021.4.0-h57928b3_3556
>  libblas            conda-forge/win-64::libblas-3.9.0-12_win64_mkl
>  libcblas           conda-forge/win-64::libcblas-3.9.0-12_win64_mkl
>  liblapack          conda-forge/win-64::liblapack-3.9.0-12_win64_mkl
>  mkl                conda-forge/win-64::mkl-2021.4.0-h0e2418a_729
>  numpy              conda-forge/win-64::numpy-1.19.5-py36h4b40d73_2
>  pip                conda-forge/noarch::pip-21.3.1-pyhd8ed1ab_0
>  python             conda-forge/win-64::python-3.6.13-h39d44d4_2_cpython
>  python_abi         conda-forge/win-64::python_abi-3.6-2_cp36m
>  setuptools         conda-forge/win-64::setuptools-49.6.0-py36ha15d459_3
>  tbb                conda-forge/win-64::tbb-2021.4.0-h2d74725_1
>  ucrt               conda-forge/win-64::ucrt-10.0.20348.0-h57928b3_0
>  vc                 conda-forge/win-64::vc-14.2-hb210afc_5
>  vs2015_runtime     conda-forge/win-64::vs2015_runtime-14.29.30037-h902a5da_5
>  wheel              conda-forge/noarch::wheel-0.37.0-pyhd8ed1ab_1
>  wincertstore       conda-forge/win-64::wincertstore-0.2-py36ha15d459_1006
>
>
>
>Downloading and Extracting Packages
>tbb-2021.4.0         | 149 KB    | ########## | 100% 
>libblas-3.9.0        | 4.5 MB    | ########## | 100% 
>numpy-1.19.5         | 4.9 MB    | ########## | 100% 
>pip-21.3.1           | 1.2 MB    | ########## | 100% 
>ucrt-10.0.20348.0    | 1.2 MB    | ########## | 100% 
>python_abi-3.6       | 4 KB      | ########## | 100% 
>certifi-2020.12.5    | 140 KB    | ########## | 100% 
>setuptools-49.6.0    | 921 KB    | ########## | 100% 
>vs2015_runtime-14.29 | 1.3 MB    | ########## | 100% 
>libcblas-3.9.0       | 4.5 MB    | ########## | 100% 
>intel-openmp-2021.4. | 3.2 MB    | ########## | 100% 
>mkl-2021.4.0         | 181.7 MB  | ########## | 100% 
>wheel-0.37.0         | 31 KB     | ########## | 100% 
>wincertstore-0.2     | 15 KB     | ########## | 100% 
>liblapack-3.9.0      | 4.5 MB    | ########## | 100% 
>python-3.6.13        | 18.9 MB   | ########## | 100% 
>vc-14.2              | 13 KB     | ########## | 100% 
>Preparing transaction: ...working... done
>Verifying transaction: ...working... done
>Executing transaction: ...working... done
>
> To activate this environment, use
>
>     $ conda activate r-reticulate
>
> To deactivate an active environment, use
>
>     $ conda deactivate
>
>* Miniconda has been successfully installed at "C:/Users/Mikip/AppData/Local/r-miniconda".
>[1] "C:/Users/Mikip/AppData/Local/r-miniconda"

`keras::install_keras()`
>Collecting package metadata (current_repodata.json): ...working... done
>Solving environment: ...working... done
>
>## Package Plan ##
>
>  environment location: C:\Users\Mikip\AppData\Local\r-miniconda\envs\r-reticulate
>
>  added / updated specs:
>    - python=3.7
>
>
>The following packages will be downloaded:
>
>    package                    |            build
    ---------------------------|-----------------
    blas-1.0                   |              mkl           6 KB
    certifi-2021.10.8          |   py37haa95532_0         151 KB
    mkl-service-2.4.0          |   py37h2bbff1b_0          49 KB
    mkl_fft-1.3.1              |   py37h277e83a_0         135 KB
    mkl_random-1.2.2           |   py37hf11a4ad_0         216 KB
    numpy-1.21.2               |   py37hfca59bb_0          24 KB
    numpy-base-1.21.2          |   py37h0829f74_0         4.4 MB
    python-3.7.11              |       h6244533_0        14.5 MB
    setuptools-58.0.4          |   py37haa95532_0         776 KB
    wincertstore-0.2           |   py37haa95532_2          15 KB
    ------------------------------------------------------------
                                           Total:        20.2 MB
>
>The following NEW packages will be INSTALLED:
>
>  blas               pkgs/main/win-64::blas-1.0-mkl
>  ca-certificates    pkgs/main/win-64::ca-certificates-2021.10.26-haa95532_2
>  mkl-service        pkgs/main/win-64::mkl-service-2.4.0-py37h2bbff1b_0
>  mkl_fft            pkgs/main/win-64::mkl_fft-1.3.1-py37h277e83a_0
>  mkl_random         pkgs/main/win-64::mkl_random-1.2.2-py37hf11a4ad_0
>  numpy-base         pkgs/main/win-64::numpy-base-1.21.2-py37h0829f74_0
>  openssl            pkgs/main/win-64::openssl-1.1.1l-h2bbff1b_0
>  six                pkgs/main/noarch::six-1.16.0-pyhd3eb1b0_0
>  sqlite             pkgs/main/win-64::sqlite-3.36.0-h2bbff1b_0
>
>The following packages will be REMOVED:
>
>  python_abi-3.6-2_cp36m
>
>The following packages will be UPDATED:
>
>  certifi                          2020.12.5-py36haa95532_0 --> 2021.10.8-py37haa95532_0
>  numpy              conda-forge::numpy-1.19.5-py36h4b40d7~ --> pkgs/main::numpy-1.21.2-py37hfca59bb_0
>  python             conda-forge::python-3.6.13-h39d44d4_2~ --> pkgs/main::python-3.7.11-h6244533_0
>  setuptools         conda-forge::setuptools-49.6.0-py36ha~ --> pkgs/main::setuptools-58.0.4-py37haa95532_0
>
>The following packages will be SUPERSEDED by a higher-priority channel:
>
>  wincertstore       conda-forge::wincertstore-0.2-py36ha1~ --> pkgs/main::wincertstore-0.2-py37haa95532_2
>
>
>
>Downloading and Extracting Packages
>setuptools-58.0.4    | 776 KB    | ########## | 100% 
>python-3.7.11        | 14.5 MB   | ########## | 100% 
>blas-1.0             | 6 KB      | ########## | 100% 
>numpy-base-1.21.2    | 4.4 MB    | ########## | 100% 
>wincertstore-0.2     | 15 KB     | ########## | 100% 
>mkl-service-2.4.0    | 49 KB     | ########## | 100% 
>mkl_random-1.2.2     | 216 KB    | ########## | 100% 
>numpy-1.21.2         | 24 KB     | ########## | 100% 
>mkl_fft-1.3.1        | 135 KB    | ########## | 100% 
>certifi-2021.10.8    | 151 KB    | ########## | 100% 
>Preparing transaction: ...working... done
>Verifying transaction: ...working... done
>Executing transaction: ...working... done
>WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.
>WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow/
>WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow/
>WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow/
>WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow/
>WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/tensorflow/
>Could not fetch URL https://pypi.org/simple/tensorflow/: There was a problem confirming the ssl certificate: >HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/tensorflow/ (Caused by >SSLError("Can't connect to HTTPS URL because the SSL module is not available.")) - skipping
>ERROR: Could not find a version that satisfies the requirement tensorflow==2.6.* (from versions: none)
>ERROR: No matching distribution found for tensorflow==2.6.*
>WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: >HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.")) - skipping
>Error: Error installing package(s): "\"tensorflow==2.6.*\"", "\"tensorflow-hub\"", "\"scipy\"", "\"requests\"", "\"Pillow\"", "\"h5py\"", "\"pandas\""

`# test if Tensorflow installed successfully`
`tensorflow::tf$abs(1)`
>Error: Valid installation of TensorFlow not found.
>
>Python environments searched for 'tensorflow' package:
> C:\Users\Mikip\AppData\Local\r-miniconda\envs\r-reticulate\python.exe
>
>Python exception encountered:
> ModuleNotFoundError: No module named 'tensorflow'
>
>You can install TensorFlow using the install_tensorflow() function.

```# if the above failed, inspect the actual import error:```
```reticulate::import("tensorflow")```
>Error in py_module_import(module, convert = convert) : 
>  ModuleNotFoundError: No module named 'tensorflow'
```for(msg in reticulate::py_last_error()) cat(msg, "\n")```
>ModuleNotFoundError 
>No module named 'tensorflow' 
>  File "C:\Users\Mikip\OneDrive\Documents\R\win-library\4.1\reticulate\python\rpytools\loader.py", line 44, in _import_hook
>    level=level
>
>ModuleNotFoundError: No module named 'tensorflow'

**Comment by Little-Miki** [comment]
The above was ran on my personal machine.

**Comment by t-kalinowski** [comment]
Hi, it looks like installing the development version of reticulate failed because Rtools is not installed. Can you please [install Rtools](https://cran.r-project.org/bin/windows/Rtools/) and try again?

**Comment by Little-Miki** [comment]
I believe that has solved the issue. Thank you for your assistance. I will monitor the model tomorrow and see if anything else falls over. Thanks again.
